### PR TITLE
Add password file argument to secret create (handles '-' as stdin read)

### PR DESCRIPTION
**Pull request by aiordache**
Add password file argument to secret create
```
$ echo "secretpassword" | docker secret create test -
arn:aws:secretsmanager:eu-west-3:175142243308:secret:test-vDMzik
```
```
$ docker secret create test pass.txt
arn:aws:secretsmanager:eu-west-3:175142243308:secret:test-Ue0s6C

```
```
$ docker secret create test -
?  *************************************
arn:aws:secretsmanager:eu-west-3:175142243308:secret:test-Hru0Qu

```
Closes https://github.com/docker/compose-cli/issues/678

/test-ecs

### CODE REVIEW

**Comment by ndeloof** [comment]
I wonder we should align with "classic" docker CLI `docker secret create [OPTIONS] SECRET [file|-]` UX and just drop support for passing plain text secret to this command

**Comment by thaJeztah** [comment]
Yes, we should definitely deprecate/remove `--password`, passing the password as argument will lead it to be ending up in bash history, and other places.

Secrets (in the docker cli) are also not just passwords, so we should avoid the use of "password" as term

**Comment by nebuk89** [comment]
> I wonder we should align with "classic" docker CLI `docker secret create [OPTIONS] SECRET [file|-]` UX and just drop support for passing plain text secret to this command

I agree, Sebastians point is a good one and this just makes everyones life messy having people think that --password is a good thing to use

File: api/secrets/api.go [hunk 1]
```diff
@@ -35,16 +35,14 @@ type Secret struct {
...
 	Labels      map[string]string `json:"Labels"`
 	Description string            `json:"Description"`
-	username    string
-	password    string
+	content     string
 }
 
 // NewSecret builds a secret
-func NewSecret(name, username, password, description string) Secret {
+func NewSecret(name, password, description string) Secret {
```
Comment (thaJeztah) [inline api/secrets/api.go:42]:
```suggestion
func NewSecret(name, content, description string) Secret {
```

**Comment by thaJeztah** [comment]
Also wondering where the `description` is coming from? For the existing CLI we don't have that, but we do provide labels as metadata (for users to use for their own use-case / conventions)

For the existing CLI we have

- `labels` as metadata (name/value)
- `name` (`[a-zA-Z0-9-_.]`, max 64 characters
- `id` (automatically generated, _no relation_ with the secret's content, so it's not a hash (e.g.))

Note that we also have a constraint on size (maximum: 500 kb)

Limiting the size that we read may be good to do, to prevent someone doing, e.g.

```bash
cat lord-of-the-rings-4k-HDR.mkv | docker secret create mymovie -
```


There's also an option for templating (allowing secrets and configs to substitute values at runtime), but that's not likely important for now

**Comment by thaJeztah** [comment]
FWIW, I think the 64 (or 63?) character limit was to allow defining DNS schemes for resources, for which 63 characters is the maximum "label" length; https://www.ietf.org/rfc/rfc1035.txt

File: cli/cmd/secrets.go [hunk 2]
```diff
@@ -56,13 +56,16 @@ func createSecret() *cobra.Command {
...
 			if err != nil {
 				return err
 			}
-			file := args[1]
+			file := "-"
+			if len(args) == 2 {
+				file = args[1]
+			}
 			if len(file) == 0 {
 				return fmt.Errorf("Secret data source empty: %q", file)
```
Comment (chris-crone) [inline cli/cmd/secrets.go:70]: [Errors should start with lower case](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)

```suggestion
				return fmt.Errorf("secret data source empty: %q", file)
```

File: cli/cmd/secrets.go [hunk 3]
```diff
@@ -76,13 +79,11 @@ func createSecret() *cobra.Command {
 					return err
 				}
 				defer in.Close()
-
 			}
 			content, err := ioutil.ReadAll(in)
 			if err != nil {
 				return fmt.Errorf("Error reading content from %q: %v", file, err)
```
Comment (chris-crone) [inline cli/cmd/secrets.go:85]: Same here
```suggestion
				return fmt.Errorf("error reading content from %q: %v", file, err)
```

File: cli/cmd/secrets.go [hunk 4]
```diff
@@ -19,19 +19,19 @@ package cmd
...
 	"github.com/docker/compose-cli/api/client"
 	"github.com/docker/compose-cli/api/secrets"
 	"github.com/docker/compose-cli/formatter"
+	"github.com/docker/docker/pkg/system"
 )
 
 type createSecretOptions struct {
 	Label       string
-	Username    string
-	Password    string
```
Comment (chris-crone) [inline cli/cmd/secrets.go:34]: This matches the existing CLI's secrets implementation which is flat (i.e.: secret name maps to a value) but I think it will break pulling private images where we need a dictionary with username and password.

File: ecs/sdk.go [hunk 5]
```diff
@@ -524,14 +524,10 @@ func (s sdk) DeleteStack(ctx context.Context, name string) error {
 
 func (s sdk) CreateSecret(ctx context.Context, secret secrets.Secret) (string, error) {
 	logrus.Debug("Create secret " + secret.Name)
-	secretStr, err := secret.GetCredString()
-	if err != nil {
-		return "", err
-	}
-
 	response, err := s.SM.CreateSecret(&secretsmanager.CreateSecretInput{
+
```
Comment (thaJeztah) [inline ecs/sdk.go:528]: nit: stray empty line

File: cli/cmd/secrets.go [hunk 6]
```diff
@@ -73,8 +95,6 @@ func createSecret() *cobra.Command {
 		},
 	}
 
-	cmd.Flags().StringVarP(&opts.Username, "username", "u", "", "username")
-	cmd.Flags().StringVarP(&opts.Password, "password", "p", "", "password")
 	cmd.Flags().StringVarP(&opts.Description, "description", "d", "", "Secret description")
```
Comment (thaJeztah) [inline cli/cmd/secrets.go:98]: Still not sure about having a `--description` flag, unless there's an important use-case that we're addressing

File: cli/cmd/secrets.go (hunk 3 above)
Comment (thaJeztah) [reply cli/cmd/secrets.go:85]: Not a blocker, just a suggestion: I usually try to avoid "error" as part of the error message (it should be clear there's an error). Also could consider using `%w` to wrap the error, in case we want to do further processing (although not likely), for example;

```suggestion
				return fmt.Errorf("failed to read content from %q: %w", file, err)
```

**Review by thaJeztah — COMMENTED** [review]
Overall looks good (but commits should be squashed). Left some minor suggestions, and a question around the `--description` flag

**Review by ndeloof — APPROVED** [review]

File: api/secrets/api.go [hunk 7]
```diff
@@ -31,21 +31,17 @@ type Service interface {
...
 type Secret struct {
-	ID          string            `json:"ID"`
-	Name        string            `json:"Name"`
-	Labels      map[string]string `json:"Labels"`
-	Description string            `json:"Description"`
-	username    string
-	password    string
+	ID      string            `json:"ID"`
+	Name    string            `json:"Name"`
+	Tags    map[string]string `json:"Tags"`
```
Comment (thaJeztah) [inline api/secrets/api.go:36]: Why was this changed from Labels to Tags? Are these Azure or ECS-specific?

If these are the equivalent to labels, we could use `--label`, and translate them to `--tag` for the API call.
Alternatively (if they're separate from "user" labels), we could consider having a prefix (`--label com.azure.tag=foo`) 

(mostly curious) 😅

Comment (aiordache) [reply api/secrets/api.go:36]: It's ECS specific :sweat_smile: We tag most of AWS resources we create with compose and I thought it is easier to follow if I renamed Labels to Tags.

Comment (ndeloof) [reply api/secrets/api.go:36]: If the plan here is to use AWS tags to implement docker labels, then should better avoid to expose AWS terminology. 

If we go this way, I wonder how TaksDefinition's labels should be set, as it has both `Tags` and  `DockerLabels`. Maybe configured labels should be applied twice there.

Comment (aiordache) [reply api/secrets/api.go:36]: Ok, I will change back to Labels :sweat:

Comment (thaJeztah) [reply api/secrets/api.go]: Thanks! (and sorry for being a PITA!). We should definitely look if there's feature-disparity between the backends, and (where possible) re-use existing terminology to keep the UX consistent. At the same time, if we find things that will affect the usability too much, we should have a look if we can find some common (platform/backend-agnostic) way to address these

**Review by chris-crone — APPROVED** [review]
LGTM

**Review by ndeloof — APPROVED** [review]
